lib/segment_tree.rb

```diff
@@ -84,17 +78,9 @@ def inspect
   end
 
   private
-  def matches?(x, low_idx, high_idx, idx) #:nodoc:
-    low, high = @segments[low_idx], @segments[high_idx]
-    segment   = @segments[idx]
-    left      = idx > 0 && @segments[idx - 1]
-    right     = idx < @segments.size - 1 && @segments[idx + 1]
-
-    case
-      when left && low.range.begin <= x && x <= left.range.end then -1
-      when segment.range.begin <=x && x <= segment.range.end then 0
-      when right && right.range.begin <=x && x <= high.range.end then 1
-      else nil
-    end
+  def matches?(x, low_idx, idx, high_idx) #:nodoc:
```

Comment:
Yeah, not too happy with it either. but as you can see on the link above (there's a small link 'click to toggle source' in the top right corner when you hover with the mouse), this whole algorithm is a duplication of what's already in bsearch. so maybe use that one specialize on the specialities for ranges, like inclusive vs. exclusive boundaries. wdyt?